io-engine/src/lvs/lvol_snapshot.rs

```diff
@@ -691,4 +713,44 @@ impl SnapshotOps for Lvol {
             .filter(|b| b.is_snapshot_clone().is_some())
             .collect::<Vec<Lvol>>()
     }
+    /// Check if the snapshot has been discarded.
+    fn is_discarded_snapshot(&self) -> bool {
+        Lvol::get_blob_xattr(self, SnapshotXattrs::DiscardedSnapshot.name())
+            .unwrap_or_default()
+            .parse()
+            .unwrap_or_default()
+    }
+
+    /// During destroying the last linked cloned, if there is any fault
+    /// happened, it is possible that, last clone can be deleted, but linked
+    /// snapshot marked as discarded still present in the system. When
+    /// control plane does the redo of the destroy replica, clone uuid will
+    /// not be found in the system, but discarded snapshot can be searched
+    /// and destroyed.
+    async fn destroy_pending_discarded_snapshot(clone_uuid: &str) {
+        info!("HR: Pending destroy: {:?}", clone_uuid);
+        let bdev = match UntypedBdev::bdev_first() {
+            Some(b) => b,
+            None => {
+                info!("HR: No Device");
+                return; /* No devices available */
+            }
+        };
+        let discarded_snapshots = bdev
+            .into_iter()
+            .filter(|b| b.driver() == "lvol")
+            .map(|b| Lvol::try_from(b).unwrap())
+            .filter(|b| {
+                b.is_snapshot()
+                    && b.is_discarded_snapshot()
+                    && b.list_clones_by_snapshot_uuid().is_empty()
+            })
+            .collect::<Vec<Lvol>>();
+        for snapshot in discarded_snapshots {
```

Comment:
I'd perform parallel future execution for the sake of performance:
```
use futures::future::join_all;
```

Reply:
Sure. Thanks for the comment. Fix it now.